### PR TITLE
feat(issues): add support for exporting issues to SARIF

### PR DESCRIPTION
**Pull request by burntcarrot**
Related: #95 

* add SARIF exporter

Signed-off-by: burntcarrot <aadhav.n1@gmail.com>

### CODE REVIEW

File: command/issues/list/testdata/test.sarif [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+            }
+          ]
+        }
+      ]
+    },
+    {
+      "tool": {
+        "driver": {
+          "informationUri": "https://deepsource.io/directory/analyzers/go",
+          "name": "DeepSource Go Analyzer",
```
Comment (siddhant-deepsource) [inline command/issues/list/testdata/test.sarif:56]: @burntcarrot Shouldn't there be a single entry for `Go` analyzer (the above one)? Ideally that should contain all the issues reported by the Go analyzer. 🤔

Comment (burntcarrot) [reply command/issues/list/testdata/test.sarif:56]: Fixing it.

File: command/issues/list/list.go [hunk 2]
```diff
@@ -281,3 +285,108 @@ func convertCSV(issueData []issues.Issue) [][]string {
...
+			shortcodes[issue.Analyzer.Shortcode] = boolIndex{exists: true, index: count}
+			count += 1
+		}
+	}
+
+	// use an index map for updating rule index value
+	idxMap := make(map[int]int)
+
+	for _, issue := range issueData {
+		textDescription := ""
```
Comment (siddhant-deepsource) [inline command/issues/list/list.go:349]:
```suggestion
        // TODO: Fetch issue description from the API and populate here
		textDescription := ""
```

File: command/issues/list/list.go [hunk 3]
```diff
@@ -281,3 +285,108 @@ func convertCSV(issueData []issues.Issue) [][]string {
...
+	// use a map of shortcodes to append rules and results
+	type boolIndex struct {
+		exists bool
+		index  int
+	}
+	shortcodes := make(map[string]boolIndex)
+	var runs []*sarif.Run
+	count := 0
+
+	for _, issue := range issueData {
```
Comment (siddhant-deepsource) [inline command/issues/list/list.go:324]:
```suggestion
    // Adding the tools data to the SARIF report corresponding to the number of analyzers activated
	for _, issue := range issueData {
```

File: command/issues/list/list.go [hunk 4]
```diff
@@ -281,3 +285,108 @@ func convertCSV(issueData []issues.Issue) [][]string {
...
+			// update boolIndex
+			shortcodes[issue.Analyzer.Shortcode] = boolIndex{exists: true, index: count}
+			count += 1
+		}
+	}
+
+	// use an index map for updating rule index value
+	idxMap := make(map[int]int)
+
+	for _, issue := range issueData {
```
Comment (siddhant-deepsource) [inline command/issues/list/list.go:348]:
```suggestion
    // Adding the results data for each analyzer in the report
	for _, issue := range issueData {
```

**Review by siddhant-deepsource — DISMISSED** [review]
Just some minor nits. Otherwise, looks good. Great work @burntcarrot 👏

**Review by siddhant-deepsource — APPROVED** [review]

**Pull request merged**